src/catalog/src/durable/impls/persist/state_update.rs

```diff
@@ -0,0 +1,624 @@
+            introspection_sources,
+            id_allocator,
+            configs,
+            settings,
+            timestamps,
+            system_gid_mapping,
+            system_configurations,
+            default_privileges,
+            system_privileges,
+            audit_log_updates,
+            storage_usage_updates,
+            // Persist implementation does not use the connection timeout.
+            connection_timeout: _,
+        } = txn_batch;
+        let databases = from_batch(databases, ts, StateUpdateKind::Database);
+        let schemas = from_batch(schemas, ts, StateUpdateKind::Schema);
+        let items = from_batch(items, ts, StateUpdateKind::Item);
+        let comments = from_batch(comments, ts, StateUpdateKind::Comment);
+        let roles = from_batch(roles, ts, StateUpdateKind::Role);
+        let clusters = from_batch(clusters, ts, StateUpdateKind::Cluster);
+        let cluster_replicas = from_batch(cluster_replicas, ts, StateUpdateKind::ClusterReplica);
+        let introspection_sources = from_batch(
+            introspection_sources,
+            ts,
+            StateUpdateKind::IntrospectionSourceIndex,
+        );
+        let id_allocators = from_batch(id_allocator, ts, StateUpdateKind::IdAllocator);
+        let configs = from_batch(configs, ts, StateUpdateKind::Config);
+        let settings = from_batch(settings, ts, StateUpdateKind::Setting);
+        let timestamps = from_batch(timestamps, ts, StateUpdateKind::Timestamp);
+        let system_object_mappings =
+            from_batch(system_gid_mapping, ts, StateUpdateKind::SystemObjectMapping);
+        let system_configurations = from_batch(
+            system_configurations,
+            ts,
+            StateUpdateKind::SystemConfiguration,
+        );
+        let default_privileges =
+            from_batch(default_privileges, ts, StateUpdateKind::DefaultPrivilege);
+        let system_privileges = from_batch(system_privileges, ts, StateUpdateKind::SystemPrivilege);
+        let audit_logs = from_batch(audit_log_updates, ts, StateUpdateKind::AuditLog);
+        let storage_usage_updates =
+            from_batch(storage_usage_updates, ts, StateUpdateKind::StorageUsage);
+
+        databases
+            .chain(schemas)
+            .chain(items)
+            .chain(comments)
+            .chain(roles)
+            .chain(clusters)
+            .chain(cluster_replicas)
+            .chain(introspection_sources)
+            .chain(id_allocators)
+            .chain(configs)
+            .chain(settings)
+            .chain(timestamps)
+            .chain(system_object_mappings)
+            .chain(system_configurations)
+            .chain(default_privileges)
+            .chain(system_privileges)
+            .chain(audit_logs)
+            .chain(storage_usage_updates)
+            .collect()
+    }
+}
+
+/// The contents of a single state update.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum StateUpdateKind {
```

Comment:
IIUC the entire catalog is written to a single persist shard, and we determine what collection something exists in via this enum? If this is true, it might be nice to add it in an enum

Reply:
Added a comment.